src/arrow_reader.rs

```diff
@@ -93,28 +117,16 @@ impl<R: Read + Seek> Iterator for ArrowReader<R> {
 
     fn next(&mut self) -> Option<Self::Item> {
         match self.current_stripe.as_mut() {
-            Some(stripe) => stripe
-                .next()
-                .map(|batch| batch.map_err(|err| ArrowError::ExternalError(Box::new(err)))),
-            None => match self
-                .cursor
-                .next()
-                .map(|r| r.map_err(|err| ArrowError::ExternalError(Box::new(err))))
-            {
-                Some(Ok(stripe)) => {
-                    match NaiveStripeDecoder::new(stripe, self.schema_ref.clone(), self.batch_size)
-                        .map_err(|err| ArrowError::ExternalError(Box::new(err)))
-                    {
-                        Ok(decoder) => {
-                            self.current_stripe = Some(Box::new(decoder));
-                            self.next()
-                        }
-                        Err(err) => Some(Err(err)),
-                    }
+            Some(stripe) => {
+                match stripe
+                    .next()
+                    .map(|batch| batch.map_err(|err| ArrowError::ExternalError(Box::new(err))))
+                {
+                    Some(rb) => Some(rb),
+                    None => self.try_advance_stripe(),
```

Comment:
Bug was here as after stripe returned None, iterator would not advance any further

Reply:
Good catch!